src/components/form/fields/identifierField.jsx

```diff
@@ -141,14 +144,15 @@ export default function identifierField(props, config) {
         },
         validator: new Validator({
             rule: value => specializeRefinedIdentifier(value,
-                v => v.isValid,
+                v => v.isValid || !props.withPhoneNumber,
```

Comment:
Adapt the validator to accept a phone number as an identifier or not depending on whether or not the feature sms is enabled and showIdentifier is set to true